# Project 1: SAT & ACT Analysis

The first markdown cell in a notebook is a great place to provide an overview of your entire project. You will likely want to at least state your

## Problem Statement

## Descriptive and Inferential Statistics

#### Summarizing Distributions

Above, we used pandas `describe` to provide quick summary statistics of our numeric columns. We also demonstrated many visual relationships.

As data scientists, having a complete understanding of data is imperative prior to modeling.

While we will continue to build our analytic tools, we know that measures of *central tendency*, *spread*, and *shape/skewness* provide a quick summary of distributions.

In [10]:
#importing the usual culprits

import pandas as pd
import numpy as np

In [11]:
#reading in our final.csv merged file:

final = pd.read_csv('/Users/grahamlim/Documents/DSI15/lab_projects_copy/project_1/data/final.csv')

#removing the pesky unnamed column again:
final.drop(['Unnamed: 0'], axis=1, inplace = True)

In [12]:
final.head()

state  2017_act_participation  2017_act_english  2017_act_math  \
0     Alabama                     100              18.9           18.4   
1      Alaska                      65              18.7           19.8   
2     Arizona                      62              18.6           19.8   
3    Arkansas                     100              18.9           19.0   
4  California                      31              22.5           22.7   

   2017_act_reading  2017_act_science  2017_act_composite  \
0              19.7              19.4                19.2   
1              20.4              19.9                19.8   
2              20.1              19.8                19.7   
3              19.7              19.5                19.4   
4              23.1              22.2                22.8   

   2017_sat_participation  2017_sat_reading_writing  2017_sat_math  ...  \
0                       5                       593            572  ...   
1                      38                       547            533  ...   
2                      30                       563            553  ...   
3                       3                       614            594  ...   
4                      53                       531            524  ...   

   2018_act_participation  2018_act_composite  2018_act_english  \
0                     100                19.1              18.9   
1                      33                20.8              19.8   
2                      66                19.2              18.2   
3                     100                19.4              19.1   
4                      27                22.7              22.5   

   2018_act_math  2018_act_reading  2018_act_science  2018_sat_participation  \
0           18.3              19.6              19.0                       6   
1           20.6              21.6              20.7                      43   
2           19.4              19.5              19.2                      29   
3           18.9              19.7              19.4                       5   
4           22.5              23.0              22.1                      60   

   2018_sat_reading_writing  2018_sat_math  2018_sat_total  
0                       595            571            1166  
1                       562            544            1106  
2                       577            572            1149  
3                       592            576            1169  
4                       540            536            1076  

[5 rows x 21 columns]

In [13]:
#using the describe command to table out summary statistics

final.describe()

2017_act_participation  2017_act_english  2017_act_math  \
count               51.000000         51.000000      51.000000   
mean                65.254902         20.931373      21.182353   
std                 32.140842          2.353677       1.981989   
min                  8.000000         16.300000      18.000000   
25%                 31.000000         19.000000      19.400000   
50%                 69.000000         20.700000      20.900000   
75%                100.000000         23.300000      23.100000   
max                100.000000         25.500000      25.300000   

       2017_act_reading  2017_act_science  2017_act_composite  \
count         51.000000         51.000000           51.000000   
mean          22.013725         21.449020           21.519608   
std            2.067271          1.737397            2.020695   
min           18.100000         18.200000           17.800000   
25%           20.450000         19.950000           19.800000   
50%           21.800000         21.300000           21.400000   
75%           24.150000         23.150000           23.600000   
max           26.000000         24.900000           25.500000   

       2017_sat_participation  2017_sat_reading_writing  2017_sat_math  \
count               51.000000                 51.000000      51.000000   
mean                39.803922                569.117647     556.882353   
std                 35.276632                 45.666901      47.121395   
min                  2.000000                482.000000     468.000000   
25%                  4.000000                533.500000     523.500000   
50%                 38.000000                559.000000     548.000000   
75%                 66.000000                613.000000     599.000000   
max                100.000000                644.000000     651.000000   

       2017_sat_total  2018_act_participation  2018_act_composite  \
count       51.000000               51.000000           51.000000   
mean      1126.098039               61.647059           21.486275   
std         92.494812               34.080976            2.106278   
min        950.000000                7.000000           17.700000   
25%       1055.500000               28.500000           19.950000   
50%       1107.000000               66.000000           21.300000   
75%       1212.000000              100.000000           23.550000   
max       1295.000000              100.000000           25.600000   

       2018_act_english  2018_act_math  2018_act_reading  2018_act_science  \
count         51.000000      51.000000         51.000000         51.000000   
mean          20.988235      21.125490         22.015686         21.345098   
std            2.446356       2.035765          2.167245          1.870114   
min           16.600000      17.800000         18.000000         17.900000   
25%           19.100000      19.400000         20.450000         19.850000   
50%           20.200000      20.700000         21.600000         21.100000   
75%           23.700000      23.150000         24.100000         23.050000   
max           26.000000      25.200000         26.100000         24.900000   

       2018_sat_participation  2018_sat_reading_writing  2018_sat_math  \
count               51.000000                 51.000000      51.000000   
mean                45.745098                563.686275     556.235294   
std                 37.314256                 47.502627      47.772623   
min                  2.000000                480.000000     480.000000   
25%                  4.500000                534.500000     522.500000   
50%                 52.000000                552.000000     544.000000   
75%                 77.500000                610.500000     593.500000   
max                100.000000                643.000000     655.000000   

       2018_sat_total  
count       51.000000  
mean      1120.019608  
std         94.155083  
min        977.000000  
25%       1057.500000  
50%       1098.000000  

In [16]:
#it's a bit unwieldy so its better to transpose it since we have so many columns.

final2 = final.describe().T
final2

count         mean        std    min      25%  \
2017_act_participation     51.0    65.254902  32.140842    8.0    31.00   
2017_act_english           51.0    20.931373   2.353677   16.3    19.00   
2017_act_math              51.0    21.182353   1.981989   18.0    19.40   
2017_act_reading           51.0    22.013725   2.067271   18.1    20.45   
2017_act_science           51.0    21.449020   1.737397   18.2    19.95   
2017_act_composite         51.0    21.519608   2.020695   17.8    19.80   
2017_sat_participation     51.0    39.803922  35.276632    2.0     4.00   
2017_sat_reading_writing   51.0   569.117647  45.666901  482.0   533.50   
2017_sat_math              51.0   556.882353  47.121395  468.0   523.50   
2017_sat_total             51.0  1126.098039  92.494812  950.0  1055.50   
2018_act_participation     51.0    61.647059  34.080976    7.0    28.50   
2018_act_composite         51.0    21.486275   2.106278   17.7    19.95   
2018_act_english           51.0    20.988235   2.446356   16.6    19.10   
2018_act_math              51.0    21.125490   2.035765   17.8    19.40   
2018_act_reading           51.0    22.015686   2.167245   18.0    20.45   
2018_act_science           51.0    21.345098   1.870114   17.9    19.85   
2018_sat_participation     51.0    45.745098  37.314256    2.0     4.50   
2018_sat_reading_writing   51.0   563.686275  47.502627  480.0   534.50   
2018_sat_math              51.0   556.235294  47.772623  480.0   522.50   
2018_sat_total             51.0  1120.019608  94.155083  977.0  1057.50   

                             50%      75%     max  
2017_act_participation      69.0   100.00   100.0  
2017_act_english            20.7    23.30    25.5  
2017_act_math               20.9    23.10    25.3  
2017_act_reading            21.8    24.15    26.0  
2017_act_science            21.3    23.15    24.9  
2017_act_composite          21.4    23.60    25.5  
2017_sat_participation      38.0    66.00   100.0  
2017_sat_reading_writing   559.0   613.00   644.0  
2017_sat_math              548.0   599.00   651.0  
2017_sat_total            1107.0  1212.00  1295.0  
2018_act_participation      66.0   100.00   100.0  
2018_act_composite          21.3    23.55    25.6  
2018_act_english            20.2    23.70    26.0  
2018_act_math               20.7    23.15    25.2  
2018_act_reading            21.6    24.10    26.1  
2018_act_science            21.1    23.05    24.9  
2018_sat_participation      52.0    77.50   100.0  
2018_sat_reading_writing   552.0   610.50   643.0  
2018_sat_math              544.0   593.50   655.0  
2018_sat_total            1098.0  1204.00  1298.0

For each variable in your data, summarize the underlying distributions (in words & statistics)
 - Be thorough in your verbal description of these distributions.
 - Be sure to back up these summaries with statistics.

## Answers:

(rounded to one decimal place)

1)2017_act_participation: The mean of 65.3 is less than the median of 69, indicating a negatively skewed distribution.

2)2017_act_english: The mean of 20.9 is slightly more than the median of 20.7, indicating a positively skewed distribution.

3)2017_act_math: The mean of 21.1 is more than the median of 20.9, indicating a positive skewed distribution.

4)2017_act_reading: The mean of 22.0 is more than the median of 21.8, indicating a positively skewed distribution. 

5)2017_act_science: The mean of 21.4 is less than the median of 21.3, indicating a slightly negatively skewed distribution.  

6)2017_act_composite: The mean of 21.5 is more than the median of 21.4, indicating a slightly positively skewed distribution  

7)2017_sat_participation: The mean of 39.8 is more than the median of 38.0, indicating a positively skewed distribution  

8)2017_sat_reading_writing: The mean of 569.1 is more than the median of 533.5, indicating a positively skewed distribution  

9)2017_sat_math: The mean of 556.9 is more than the median of 548.0, indicating a positively skewed distribution  

10)2017_sat_total: The mean of 1126.1 is more than the median of 1107.0, indicating a positively skewed distribution  

11)2018_act_participation: The mean of 61.6 is less than the median of 66.0, indicating a negatively skewed distribution  

12)2018_act_composite: The mean of 21.5 is more than the median of 21.3, indicating a slightly positively skewed distribution  

13)2018_act_english: The mean of 21.0 more than the median of 20.2, indicating a positively skewed distribution  

14)2018_act_math: The mean of 21.1 is more than the median of 20.7, indicating a positively skewed distribution  

15)2018_act_reading: The mean of 22.0 is more than the median of 21.6, indicating a positively skewed distribution  

16)2018_act_science: The mean of 21.3 is more than the median of 21.1, indicating a slightly positively skewed distribution  

17)2018_sat_participation: The mean of 45.7 is less than the median of 52.0, indicating a negatively skewed distribution  

18)2018_sat_reading_writing: The mean of 563.7 is more than the median of 552.0, indicating a positively skewed distribution  

19)2018_sat_math: The mean of 556.2 is more than the median of 544.0, indicating a positively skewed distribution  

20)2018_sat_total: The mean of 1120.0 is more than the median of 1098.0, indicating a positively skewed distribution  

#### We generally assume that data we sample from a population will be normally distributed. Do we observe this trend?

### Answer: 
Not in this case. Technically, every single one of our variables above are skewed. A true normal distribution has zero skewness, but outside of a strict definition of normal distribution, most of the variables above have medians that do not stray too far away from the mean e.g. in 2018_act_science - the mean and median has only got a slightly positively skewed distribution.

#### Does This Assumption Hold for:
    - Math
    - Reading
    - Rates
Explain your answers for each distribution and how you think this will affect estimates made from these data.

### Answer: 

The assumption of normal distributions still does not work with Math and English scores. All math and english for either test in both years are skewed positively, although the median and the mean do tend quite close to each other.

The assumption of normal distributions definitely does not work with Participation Rates, though. 

ACT Participation rates are negatively skewed both years.

SAT participation rates, however, switch from being positively skewed to negatively skewed over the 2 years - which is reflective of the sudden spike in 100% participation rates in that test in 2018.

Overall, all estimates made from our data therefore will all have long tails i.e significant outliers will influence any statistics, hence the skewness.

#### Estimate Limits of Data

Suppose we only seek to understand the relationship between SAT and ACT participation rates in 2017. 

##### Does it make sense to conduct statistical inference given these data specifically? 

Why or why not?

*(think about granularity, aggregation, the relationships between populations size & rates...consider the actually populations these data describe in answering this question)*

### Answer: 

It is not completely sensible to completely rely on statistical inference for this data. This is because we don't actually what the population weightage is per State - it could be that regardless of lower participation rates in the SAT in 2017, they may still have a large absolute number of test-takers, because we only know the percentages, not the raw absolute numbers of test-takers in either test. 

We also find that some states force their students to take the SAT the very next year in 2018, as is the case with Colorado and Illinois. in 2017, the SAT was optional still - so if we try to use both years to find purely data-driven insights about these States without researching the context, statistical inference will be less useful.

Source: https://chicago.chalkbeat.org/2018/7/27/21105418/illinois-has-embraced-the-sat-and-the-act-is-mad-about-it

##### Is it appropriate to compare *these* specific SAT and ACT math scores? 

Why or why not?

### Answer: 

It's not really appropriate. The tests are not equivalent and are scored on totally different grading scales (ACT aggregates average subject scores; SAT simply adds up an English and Math score). For example, unlike the ACT the SAT has a section whereby the use of calculators is omitted: https://blog.prepscholar.com/act-vs-sat

## Outside Research

Based upon your observations, choose **three** states that demonstrate interesting trends in their SAT and/or ACT participation rates. Spend some time doing outside research on state policies that might influence these rates, and summarize your findings below. **Feel free to go back and create new plots that highlight these states of interest**. If you bring in any outside tables or charts, make sure you are explicit about having borrowed them. If you quote any text, make sure that it renders as being quoted. (Make sure that you cite your sources -- check with you local instructor for citation preferences).

In [28]:
final[['state','2017_sat_participation','2018_sat_participation', '2017_act_participation', '2018_act_participation']].head(5)

state  2017_sat_participation  2018_sat_participation  \
0     Alabama                       5                       6   
1      Alaska                      38                      43   
2     Arizona                      30                      29   
3    Arkansas                       3                       5   
4  California                      53                      60   

   2017_act_participation  2018_act_participation  
0                     100                     100  
1                      65                      33  
2                      62                      66  
3                     100                     100  
4                      31                      27

## Answer:

## The 3 States of Interest are Colorado, Illinois and Alaska.

From 2017-2018, Colorado and Illinois show a gigantic jump in participation (from our previous Exploratory Data Analysis - we see that Colorado had a massive rise in participation from 11% to 100%,
Significantly, though not up to 100%, Illinois increased SAT participation from 9% to 99% (11 times more).

Alaska has got consistently low participation rates generally in 2018 for both tests. Though that seems unremarkable, it is of interest to us because the state had an almost 50% drop in ACT participation rates between 2017-2018, but unlike Colorado and Illinois, this was NOT followed by a huge spike in SAT participation in 2018 - it rose only by a mere 43%. This is worth investigating as a potential target State for the SAT to try to improve participation rates in.

## Research and Reasoning

The big question is why Colorado and Illinois had their participation rates shoot up. In both cases, the answer is successful state-wide mandatory SAT test-taking contracts were won by The College Board.

Illinois' Education Board "weighed the attributes of both exams and decided the SAT was better aligned with standards for what students should know in Illinois.":https://www.chicagotribune.com/news/ct-illinois-chooses-sat-met-20160211-story.html. Similarly, Colorado picked the SAT because of "strong alignment to Colorado's academic standards and its free practice tools.": https://collegereadiness.collegeboard.org/state-partnerships/colorado. Thus, the research indicates that the SAT looks like a more appropriate test from an academic suitability argument.

In fact, the SAT test's superiority is so compelling, that Illinois picked the SAT over the ACT despite the latter bidding 5.7m USD cheaper than The College Board's bid: https://chicago.chalkbeat.org/2018/7/27/21105418/illinois-has-embraced-the-sat-and-the-act-is-mad-about-it


Colorado and Illinois are certainly strong wins for SAT, but our dataset shows that the ACT is apparently ahead in participation rates (17 states with 100% participation). Recall our boxplot of participation rates from Data Visualization.

However, percentage rates are misleading - as they only show participation rates in individual States. 

In terms of the absolute number of test-takers, SAT's successful contracts with Illinois and Colorado mandating SAT tests were 'instrumental' allowing SAT to overtake ACT as the market leader. As a result, in 2018, SAT test-takers rose to 2m, vs the ACT's 1.91m in 2018:
https://www.washingtonpost.com/education/2018/10/23/sat-reclaims-title-most-widely-used-college-admission-test/. This is of little surprise given that Illinois is home to Chicago, the 3rd largest US City by population: https://en.wikipedia.org/wiki/List_of_United_States_cities_by_population.

If the research above shows that the SAT is the better test, then states who appear "on the fence" with either test could be targets for State SAT exam mandates, like Alaska (whom we picked as a state to investigate). 


## Conclusions and Recommendations

Based on your exploration of the data, what are you key takeaways and recommendations? Choose one state with a lower participation rate and provide a suggestion for how the College Board might increase participation amongst graduating seniors in this state. Are there additional data you desire that would better inform your investigations?

## Answer:

Our first conclusion is that participation percentage rates aren't everything - we must look at the absolute number of test takers too, given our revelation that the SAT has, in fact, been gaining ground since 2017 - which our data also showed in terms of the jump in participation rates. Research shows that in 2018, the SAT already overtook the ACT in terms of absolute test-taker numbers. How can we keep propelling this success?

A first recommendation is to continue to go after States that have very high ACT participation rates. Repeating the success of SAT 'converts' like Colorado or Illinois will clearly improve participation rates, and obviously increase the absolute number of participants - because the students will have no choice. The College Board would literally be stealing away market share from The ACT.
Still, government contracts typically have high barriers to entry. Thus, while it would be great to replicate Colorado again, we may look at lower-hanging fruit.

Our second recommendation, then, would be to market the test directly to students in states who are 'on the fence' with participating in either test. To use Alaska as an example, we already know their participation rates were less than 50% in either test in 2018, indicating ambivalence to either test. 

Our data also showed that States who had greater than 50% participation in both tests would tend to score poorly (see Exploratory Data Analysis), indicating that perhaps the students were overstretched in attempting two different exams. If it can be marketed to the students of "ambivalent" and "overstretched" States that statistically, students do better if they focus on the one test, why not share the Colorado/Illinois success story switching to the one test that was reportedly most appropriate - the SAT?

Nevertheless, any business plan to market towards students in any given State must not be made without contextual research. Alaska, as it turns out, would be a futile target for the SAT. They stopped all mandatory testing :https://www.adn.com/alaska-news/education/2016/06/30/students-no-longer-need-national-tests-to-graduate/. This happened in 2016, which partly explains the weak participation rates in our data generally.

Again, to highlight the importance of context, tests in general have reduced importance in places like Alaska. Its economy is driven mostly by low-tech industries, reducing the need for college education. According to Wikipedia, Alaska is "the third-least populous and the most sparsely populated state... economy dominated by fishing, natural gas, and oil". The US military also contribute significantly to the economy by employing many Alaskans: https://en.wikipedia.org/wiki/Alaska

In summary, with this multi-pronged approach, and with considered thought into context, we would have a higher probability of furthering the ongoing success of the SAT.
